cloud/ovirt/ovirt_templates_facts.py

```
@@ -0,0 +1,95 @@
+author: "Ondra Machacek (@machacekondra)"
+version_added: "2.3"
+description:
+    - "Retrieve facts about one or more oVirt templates."
+notes:
+    - "This module creates a new top-level C(ovirt_templates) fact, which
+       contains a list of templates."
+options:
+    pattern:
+      description:
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search template X from datacenter Y use following pattern:
+           name=X and datacenter=Y"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all templates which names start with C(centos) and
+# belongs to data center C(west):
+- ovirt_templates_facts:
+    pattern: name=centos* and datacenter=west
+- debug:
+    var: ovirt_templates
+'''
+
+RETURN = '''
+ovirt_templates:
+    description: "List of dictionaries describing the templates. Template attribues are mapped to dictionary keys,
+                  all templates attributes can be found at following url: https://ovirt.example.com/ovirt-engine/api/model#types/template."
+    returned: On success.
+    type: list
+'''
+
+
+def main():
+    argument_spec = ovirt_full_argument_spec(
+        pattern=dict(default='', required=False),
+    )
+    module = AnsibleModule(argument_spec)
+    check_sdk(module)
+
+    try:
+        connection = create_connection(module.params.pop('auth'))
+        templates_service = connection.system_service().templates_service()
+        templates = templates_service.list(search=module.params['pattern'])
+        module.exit_json(
+            changed=False,
+            ansible_facts=dict(
+                ovirt_templates=[
+                    get_dict_of_struct(c) for c in templates
+                ],
+            ),
+        )
+    except Exception as e:
+        module.fail_json(msg=str(e))
```

Same here, please surface the full exception.